\documentclass[12pt,letterpaper]{article}
\usepackage[margin=1in]{geometry}
\begin{document}

\title{GEO Dataset Tools Documentation}
\author{Michael A. Gohde}
\maketitle

\section{Introduction}
The GEO Dataset Tools provide a set of utilities and functions that should make it relatively straightforward to 
query the NIH's GEO database. Ultimately, the information gained from these tools should make it easier to download
datasets and to collect information about how they were created.

Currently, the tools present here are in a state of fairly heavy development, so any discrepancies between that which
is documented here and the actual operation of the tools is likely due to recent changes to how they were designed.

\section{Querying the GEO Database for Entries}
The first step in the process of finding specific datasets to work with is to actually determine which datasets exist. 
In the project directory, there exists a script named, `query\_geo.py'. Its purpose is to search the GEO dataset for various
terms related to the protocols used for data collection and analysis. For example, it can be used to determine which entries
in the GEO database were generated using GRO-Seq.

When the script is invoked, it queries the GEO database with the specified search term or, ``gro-seq'' if none is specified. If the
query is successful, it will proceed to either print or write a file containing a set of database element IDs corresponding to
elements matching the query. With this set of IDs, it will be possible to invoke another script to fetch all of the found elements from GEO.
This process will be detailed in the next section.

\subsection{Invoking query\_geo.py}
At the time of this writing, query\_geo can be invoked in the following ways:

\begin{verbatim}
Usage: ./query_geo.py <args> query
Query the GEO database for numbers usable by fetch_groseq.py
If the -o switch is not specified, it is recommended that this program's output
be piped into either another command or a file.

<args> may be one of the following:
-h, --help      Prints this message.
-o=<filename>   Writes to an output file instead of stdout.
-v      Verbose output. All logging messages are written to stderr.

Examples:
./query_geo.py gro-seq >out.txt
./query_geo.py -o=out.txt pro-seq
\end{verbatim}

Generally, the query parameter will correspond to a specific protocol, such as pro-seq, gro-seq, gro-cap, and 5'gro. In fact, the 
database fetching software detailed later in this document expects that this is the case. However, it should be possible to 
query the database for other features of interest, such as specific author names, species, etc.

It should also be noted that all this script works best when its output is redirected to a separate file. This makes it easier to view
debugging information when the `-v' parameter is specified. Also, the other scripts in this project expect to be able to read a file instead of
receiving GEO database IDs from the command line. 

Below is an example session used to demonstrate a series of queries made to the GEO database:
\begin{verbatim}
user@computer ~/geo_dataset_tools $ ./query_geo.py -o=groseq.txt -v gro-seq
[Message] About to make query...
[Message] Query successful!
[Message] Found 1373 elements.
[Message] Done.

user@computer ~/geo_dataset_tools $ ./query_geo.py -o=proseq.txt -v pro-seq
[Message] About to make query...
[Message] Query successful!
[Message] Found 106 elements.
[Message] Done.

user@computer ~/geo_dataset_tools $ ./query_geo.py -o=5gro.txt "5'gro"
\end{verbatim}

Note that the last command didn't print anything to the console. This is so because the `-v' parameter was not specified.

Contents of an the `5gro.txt' file generated above:
\begin{verbatim}
QUERY 5'gro
200068677
200090035
200083108
200045914
301678910
301678908
302396016
302396015
302193123
301119600
301119599
\end{verbatim}

\section{Generating a local metadata database}
Once one or more sets of IDs has been obtained following the procedure outlined in the previous section, it's necessary to 
generate and maintain a local cache of information from the GEO database. This is done for a number of reasons:
    
\begin{enumerate}
 \item It's significantly faster than repeatedly querying an online database.
 \item Keeping a local cache reduces the load on the GEO database servers.
 \item Having a local database following strict formatting rules will enable more applications to be developed in the future.
 \item Additional data can be associated with each element, such as data series matrices, etc.
 \item This enables consistency in implementation and usage.
\end{enumerate}

In order to create this database, all that's necessary is to run the following command for all of the ID files fetched in the previous section:
\begin{verbatim}
user@computer ~/geo_dataset_tools $ ./make_groseq_database.py groseq.txt 
    proseq.txt 5seq.txt grocap.txt db
\end{verbatim}


In the above example, the local database creation tool will read through every ID defined in all of the files specified, then store appropriate metadata in
the directory specified. It is possible to specify any number of input files (as long as there is at least one), and more queries can be integrated into the
same database directory after it is populated for the first time. In fact, it should also be possible to merge the same query back into the database at a future
date. Doing so will allow any changed elements to be added, while existing elements will be updated if changes have been made.

Full usage statement for the make\_groseq\_database script:
\begin{verbatim}
Usage: ./make_groseq_database.py idfile(s) dbdir
Fetches project summaries and adds appropriate metadata to a database of GEO 
GRO-Seq data.
WARNING: This script may generate several thousand directories under dbdir.
\end{verbatim}

\section{Querying the local metadata database}
Once enough metadata has been successfully fetched, it is possible to perform operations involving that metadata.
In order to do so, it is necessary to invoke the `query\_groseq\_database.py' script in the project directory.

The script's usage statement is listed below:
\begin{verbatim}
Usage: ./query_groseq_database.py [-s,-pt,-lq] dbdir command <args>
Query a GRO-Seq metadata database fetched with make_groseq_database.py
If -s is specified, then only series IDs will be reported on
If -pt=<comma separated list of protocols> is specified, then only IDs with a 
     specific protocol will be reported on.
If -lq or --last-query is specified, then the program will attempt to read as 
     arguments the results of the last query.

List of commands:
  listprotocols -- List all protocols in the current database.
  queryprotocol -- Print all elements matching a given protocol.
  protocoloverlap -- Print out the set of elements that overlap between different protocols.
  listspecies -- Print a listing of all species defined in the database.
  findspecies <species name in quotes> -- Find all projects that match a given species.
  getsummary <list of id numbers or paper names> -- Retrieves a summary for a given data element.
  fetchmatrices <id> -- Downloads data matrices necessary to fetch data for a set.
  fetchspmats <species name> -- Fetches all matrices for a given species.
  fetchallmatrices -- Fetch as many matrix files as possible. This will be slow!
  getsralist <id or paper name> -- Retrieves all SRAs for a given element given that matrices are present.
  getreadytosra -- Retrieves a list of all projects with fetched matrix files.
  getreadytodownload -- Retrieves a list of all projects that can be downloaded immediately.
  getbyyear <year> -- Retrieves a list of all projects with downloaded series matrices by year posted.
  getbycontrib <contributor name> -- Retrieves a list of all projects with downloaded series matrices
  listyears -- List all years appearing in data matrices.
  listcontribs -- List all contributors appearing in data matrices.
       by the first contributor.
  download <id or paper name> <outputdir> -- Downloads data into the specified directory
\end{verbatim}

\subsection{An explanation of the `-s', `-pt' and `-lq' command line arguments}
As documented, every command implemented in `query\_groseq\_database' is designed to work with one of the following sets of 
elements:
\begin{enumerate}
 \item Every element present in the database.
 \item A specific set of elements explicitly listed by the user.
 \item A specific element explicitly listed by the user.
\end{enumerate}

The first case in particular is extremely limited in that it will often present quite a bit of unwanted data for every given query. For example,
a query based on species name will include both collections of data and every entry for every data element in each collection.

The `-s' and `-pt' command line arguments reduce the size of the input set that query commands work on by allowing the user to specify specific
additional attributes for what they would like to find. As such, the `-s' command limits all queries to just the set of defined collections, rather than 
the elements making them up. The `-pt' command similarly allows users to search for data collected under specific protocols.

`-lq' works somewhat differently from `-s' and `-pt'. It is used to specify that the list of IDs or paper names normally required of certain commands like `search'
are instead to be found in either a special file created after every query or in a named file specified by the user. Instead of modifying the set of elements
to be iterated over when doing a search, it appends the contents of the file to the arguments list passed to the program.

\subsection{listprotocols}
This command lists the set of protocols defined in the database. This is actually the list of queries fed into the make\_groseq\_database script. 
At the time of this writing, this command effectively just lists the contents of the database directory specified.

\subsection{queryprotocol}
This command fetches a list of all elements matching a given protocol. This list includes a set of ID numbers and human-readable titles provided by the GEO database.

\subsection{protocoloverlap}
This command causes query\_groseq\_database to attempt to find which elements are the same across protocols. Certain data series defined by GEO may have multiple protocols
associated with them. Running protocoloverlap should make it easier to find out which datasets have which protocols associated with them. In the future, more commands
like this one will be implemented.

\subsection{listspecies}
This command does as specified. It searches for and returns a list of all species defined in the dataset sorted by the frequency in which they were mentioned.
At the time of this writing, it is noted that certain data series may specify multiple species. This situation still must be rectified.

\subsection{findspecies}
This command searches for all IDs matching the specified species. At the time of this writing, multiple species may be defined per data element. This situation 
will be resolved or otherwise noted through program behavior in the future.

\subsection{getsummary}
This command prints a detailed summary for every ID or paper name specified. Individual IDs are to be separated by spaces. Eventually, wildcard and similar
match support should be implemented.

\subsection{fetchmatrices}
This command fetches a set of data series matrices for the appropriate ID if they are defined in that ID's metadata. Please note that this command requires that series matrices
be defined for the given data element. While almost every element has an associated matrix, you can check for whether one can be fetched by checking for the "Data matrix URL defined?"
field on running the getsummary command. If "YES" is printed next to the field, then matrices can be fetched. Otherwise, no data matrices have been defined for the current element
and thus cannot be fetched.

One side effect of having fetched series matrices for a given element is that every future listing for that element by other commands will contain a ``paper name.''
This is a string consisting of the last name of the first contributor to the element's project along with the date that the element was posted to GEO. Please
note that publication years for papers and the dates at which their associated data was submitted to GEO may differ.

\subsection{fetchspmats}
This command was implemented to make it easier to fetch all data matrices for a given species. It is equivalent to running the `findspecies' command, then running
`fetchmatrices' for every result returned.

\subsection{fetchallmatrices}
This command attempts to fetch every series matrix file associated with every element in the database. Please note that due to the intensive nature of this command,
it may take a very long time to complete and lacks much formal validation.

\subsection{getsralist}
If data series matrices have been successfully fetched for a given element, then this command looks up the SRA ID numbers for every data file associated
with that element.

\subsection{getreadytosra}
This command is a convenient way of listing every project for which series matrices have been fetched.

\subsection{getreadytodownload}
This command lists all elements on which `getsralist' has been run.

\subsection{download}
This command downloads the set of SRA files found by the `getsralist' command into the directory specified on the command line.

\subsection{getbyyear}
This command lists all elements posted to GEO in the specified year. Please note that the only way to determine the year of publication is to fetch the set of 
series matrix files associated with a given entry.

\subsection{listyears}
This command searches through all data elements for which series matrices have been downloaded to produce a set of publication years and their frequencies. 
This is useful when determining the set of publication years to pass to `getbyyear'.

\subsection{getbycontrib}
This command works in much the same way as `getbyyear', except it searches by the last name of the first contributor.

\subsection{listcontribs}
This command works in much the same way as `listyears', except it produces a set of contributor names and their frequencies. Please note that it only considers
the name of the first contributor for any given project, though it may, in the future, search through all contributors.

\subsection{getaccession}
This command searches through the database for a given GEO accession number or numbers and
returns a list of all matching database IDs. 

\subsection{listsras}
This command lists all of the SRA numbers associated with the IDs provided. It assumes that
getsralist has been run for all of the IDs provided.

\subsection{examples}
This subsection shows a number of examples that should demonstrate how `query\_groseq\_database' is used in practice.
For all of the examples listed, the database is stored in a directory named `db'.

The following is an example of a simple query listing all species in the database. Please note that the `-s' flag has been specified, so
only GEO series entries are searched for.
\begin{verbatim}
user@computer-$ ./query_groseq_database.py -s db listspecies
List of available species:
125 Homo sapiens
62 Mus musculus
24 Drosophila melanogaster
10 Caenorhabditis elegans
4 Rattus norvegicus
3 Pan troglodytes
3 Macaca mulatta
2 Saccharomyces cerevisiae
2 Arabidopsis thaliana
1 Zea mays
1 Tetrahymena thermophila
1 Sus scrofa
1 Schizosaccharomyces pombe
1 Plasmodium falciparum
1 Canis lupus familiaris

Total number of elements: 241
\end{verbatim}

The following is like the above query, but only for gro-seq data:
\begin{verbatim}
user@computer-$ ./query_groseq_database.py -s -pt="gro-seq" db listspecies
List of available species:
99 Homo sapiens
57 Mus musculus
18 Drosophila melanogaster
7 Caenorhabditis elegans
2 Rattus norvegicus
1 Zea mays
1 Tetrahymena thermophila
1 Sus scrofa
1 Saccharomyces cerevisiae
1 Plasmodium falciparum
1 Pan troglodytes
1 Macaca mulatta
1 Arabidopsis thaliana

Total number of elements: 191
\end{verbatim}

The following is a listing of all entries for Rattus norvegicus. Note once again
that all non-series entries have been filtered out:
\begin{verbatim}
user@computer-$ ./query_groseq_database.py -s db findspecies "Rattus norvegicus"
Found 4 elements that match "Rattus norvegicus" given protocol(s) 5'gro,pro-seq,gro-cap,gro-seq
List of paths:

[pro-seq] 200085337: Natural Selection has Shaped Coding and Non-coding Transcription in Primate CD4+ T-cells

[pro-seq] "Mishmar2016" 200085747: Initiation of mtDNA transcription is followed by pausing, and diverge across human cell types and during evolution

[gro-seq] 200085747: Initiation of mtDNA transcription is followed by pausing, and diverge across human cell types and during evolution

[gro-seq] 200058009: Required Enhancer: Matrin-3 Structure Interactions for Homeodomain Transcription Programs
\end{verbatim}

The above query generated a hidden file named `.lastquery'. This file contains the list of IDs found in a search query. Note that the set
of IDs shown matches the set mentioned in the previous example:
\begin{verbatim}
200085337
200085747
200085747
200058009
\end{verbatim}

This is an example using the `.lastquery' file listed before. It generates a listing of summaries for all of the elements specified within the file.
It is also possible to specify additional paper names or element IDs for consideration:
\begin{verbatim}
user@computer-$ ./query_groseq_database.py -lq -s db getsummary
----Summary for 200085337----
Title: Natural Selection has Shaped Coding and Non-coding Transcription in Primate CD4+ T-cells
Posted: 2016/12/29
Accession nr: GSE85337
Species: Pan troglodytes; Rattus norvegicus; Mus musculus; Macaca mulatta; Homo sapiens
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE85nnn/GSE85337/
Summary (begins on next line):
Transcriptional regulatory changes have been shown to contribute to phenotypic differences between species, but many questions remain about how gene expression evolves. Here we report the first comparative study of nascent transcription in primates. We used PRO-seq to map actively transcribing RNA polymerases in resting and activated CD4+ T-cells in multiple human, chimpanzee, and rhesus macaque individuals, with rodents as outgroups. This approach allowed us to directly measure active transcription separately from post-transcriptional processes. We observed general conservation in coding and non-coding transcription, punctuated by numerous differences between species, particularly at distal enhancers and non-coding RNAs. Transcription factor binding sites are a primary determinant of transcriptional differences between species. We found evidence for stabilizing selection on gene expression levels and adaptive substitutions associated with lineage-specific transcription. Finally, rates of evolutionary change are strongly correlated with long-range chromatin interactions. These observations clarify the role of primary transcription in regulatory evolution.

Data matrix URL defined? YES
Ready to fetch data? NO (run fetchmatrices 200085337)
Data fetched? NO
Protocol: pro-seq

----Summary for 200085747----
Title: Initiation of mtDNA transcription is followed by pausing, and diverge across human cell types and during evolution
Posted: 2016/09/30
Accession nr: GSE85747
Species: Macaca mulatta; Homo sapiens; Caenorhabditis elegans; Drosophila melanogaster; Pan troglodytes; Rattus norvegicus; Mus musculus
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE85nnn/GSE85747/
Summary (begins on next line):
We analyzed nascent mtDNA-encoded RNA transcripts from GRO-seq and PRO-seq experiments in a collection of diverse human cell lines and Metazoan organisms. Accurate detection of human mtDNA transcription initiation sites (TIS) in the heavy and light strands revealed a novel transcription pausing site near the light strand TIS, upstream to the transcription-replication transition region, in conserved sequence block III (CSBIII). The transcription pausing index varied quantitatively among the cell lines. In addition to the human cell lines experiments, our experiments and analysis of non-human organisms enabled detection of previously unknown mtDNA TIS, pausing, and transcription termination sites with unprecedented accuracy. Whereas mammals (chimpanzee, rhesus macaque, rat, and mouse) showed a human-like mtDNA transcription pattern, the invertebrate pattern (Drosophila and C. elegans) profoundly diverged.

Data matrix URL defined? YES
Ready to fetch data? YES
Data fetched? NO
Protocol: pro-seq

----Summary for 200085747----
Title: Initiation of mtDNA transcription is followed by pausing, and diverge across human cell types and during evolution
Posted: 2016/09/30
Accession nr: GSE85747
Species: Macaca mulatta; Homo sapiens; Caenorhabditis elegans; Drosophila melanogaster; Pan troglodytes; Rattus norvegicus; Mus musculus
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE85nnn/GSE85747/
Summary (begins on next line):
We analyzed nascent mtDNA-encoded RNA transcripts from GRO-seq and PRO-seq experiments in a collection of diverse human cell lines and Metazoan organisms. Accurate detection of human mtDNA transcription initiation sites (TIS) in the heavy and light strands revealed a novel transcription pausing site near the light strand TIS, upstream to the transcription-replication transition region, in conserved sequence block III (CSBIII). The transcription pausing index varied quantitatively among the cell lines. In addition to the human cell lines experiments, our experiments and analysis of non-human organisms enabled detection of previously unknown mtDNA TIS, pausing, and transcription termination sites with unprecedented accuracy. Whereas mammals (chimpanzee, rhesus macaque, rat, and mouse) showed a human-like mtDNA transcription pattern, the invertebrate pattern (Drosophila and C. elegans) profoundly diverged.

Data matrix URL defined? YES
Ready to fetch data? YES
Data fetched? NO
Protocol: pro-seq

----Summary for 200058009----
Title: Required Enhancer: Matrin-3 Structure Interactions for Homeodomain Transcription Programs
Posted: 2014/08/03
Accession nr: GSE58009
Species: Rattus norvegicus
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE58nnn/GSE58009/
Summary (begins on next line):
Study of the POU-homeodomain transcription factor, has revealed that, binding of Pit1-occupied enhancers to a nuclear matrin-3-rich network/architecture is a key event in effective activation of the Pit1-regulated enhancer / coding gene transcriptional program.

Data matrix URL defined? YES
Ready to fetch data? NO (run fetchmatrices 200058009)
Data fetched? NO
Protocol: gro-seq
\end{verbatim}

The following is an example of a query for which the user wishes to store results in a specific file:
\begin{verbatim}
user@computer-$ ./query_groseq_database.py -qf=query.txt -s db getbyyear 2015
[gro-seq] "Yu2015" 200071369: Panoramix enforces piRNA-dependent co-transcriptional silencing (GRO-Seq)

[gro-seq] "Fuda2015" 200058955: GAGA Factor maintains promoters in nucleosome-free conformation and allows promoter-proximal pausing [GRO-seq]

[gro-seq] "Shilatifard2015" 200070408: PAF1, a molecular regulator of promoter-proximal pausing by RNA Polymerase II

[gro-seq] "Fuda2015" 200058956: GAGA Factor maintains promoters in nucleosome-free conformation and allows promoter-proximal pausing [MNase]

[gro-seq] "Fuda2015" 200058957: GAGA Factor maintains promoters in nucleosome-free conformation and allows promoter-proximal pausing
\end{verbatim}

Once again, summaries are fetched given the query file specified:
\begin{verbatim}
user@computer-$ ./query_groseq_database.py -lq=query.txt -s db getsummary
----Summary for 200071369----
Title: Panoramix enforces piRNA-dependent co-transcriptional silencing (GRO-Seq)
Posted: 2015/10/15
Accession nr: GSE71369
Species: Drosophila melanogaster
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE71nnn/GSE71369/
Summary (begins on next line):
The Piwi-interacting RNA (piRNA) pathway is a small RNA-based innate immune system that defends germ cell genomes against transposons. In Drosophila ovaries, the nuclear Piwi protein is required for transcriptional silencing of transposons, though the precise mechanisms by which this occurs are unknown. Here we show that CG9754 is a component of Piwi complexes that functions downstream of Piwi and its binding partner, Asterix, in transcriptional silencing. Enforced tethering of CG9754 protein to nascent mRNA transcripts causes co-transcriptional silencing of the source locus and the deposition of repressive chromatin marks. We have named CG9754 Panoramix, and propose that this protein could act as an adaptor, scaffolding interactions between the piRNA pathway and the general silencing machinery that it recruits to enforce transcriptional repression.

Data matrix URL defined? YES
Ready to fetch data? YES
Data fetched? NO
Protocol: gro-seq

----Summary for 200058955----
Title: GAGA Factor maintains promoters in nucleosome-free conformation and allows promoter-proximal pausing [GRO-seq]
Posted: 2015/04/04
Accession nr: GSE58955
Species: Drosophila melanogaster
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE58nnn/GSE58955/
Summary (begins on next line):
Promoter-proximal pausing of RNA polymerase II (Pol II) is a widespread in higher eukaryotes. Previous studies have shown that GAF is enriched at paused genes, but the role of GAF in pausing has not been well characterized on a genome-wide level. To investigate the role of GAF in pausing, we RNAi-depleted GAF from Drosophila S2 cells, and examined the effects on promoter-proximal polymerase. We confirmed the importance of GAF for pausing on the classic pause model gene Hsp70. To determine the dependence of pausing on GAF genome-wide, we assayed the levels of transcriptionally-engaged polymerase genome-wide using GRO-seq in control and GAF-RNAi cells. We found that promoter-proximal polymerase was significantly reduced on a subset of paused genes with GAF-bound promoters. There is a dramatic change in nucleosome distribution at genes with reduction in pausing upon GAF depletion and intergenic GAF binding sites in GAF knock-down, suggesting that GAF allows the establishment of pausing at these genes by directing nucleosome displacement off of the promoter. In addition, the insulator factor BEAF, BEAF-interacting protein Chriz, and transcription M1BP enrichment on unaffected genes suggests that redundant transcription factors or insulators protect other GAF-bound paused genes from GAF knock-down effects.

Data matrix URL defined? YES
Ready to fetch data? YES
Data fetched? NO
Protocol: gro-seq

----Summary for 200070408----
Title: PAF1, a molecular regulator of promoter-proximal pausing by RNA Polymerase II
Posted: 2015/08/13
Accession nr: GSE70408
Species: Homo sapiens; Drosophila melanogaster
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE70nnn/GSE70408/
Summary (begins on next line):
The control of promoter-proximal pausing and the release of RNA polymerase II (RNA Pol II) is a widely used mechanism for regulating gene expression in metazoans, especially for genes that respond to environmental and developmental cues. Here, we identify Pol II associated Factor 1 (PAF1) as a major regulator of promoter-proximal pausing. Knockdown of PAF1 leads to increased release of paused Pol II into gene bodies at thousands of genes. Genes with the highest levels of paused Pol II exhibit the largest redistribution of Pol II from the promoter-proximal region into the gene body in the absence of PAF1. PAF1 depletion results in increased nascent transcription and increased levels of phosphorylation of Pol IIâ€™s c-terminal domain on serine 2 (Ser2P). These changes can be explained by the recruitment of the Ser2P kinase Super Elongation Complex (SEC) effecting increased release of paused Pol II into productive elongation, thus establishing a novel function for PAF1 as a major regulator of pausing in metazoans.

Data matrix URL defined? YES
Ready to fetch data? YES
Data fetched? NO
Protocol: gro-seq

----Summary for 200058956----
Title: GAGA Factor maintains promoters in nucleosome-free conformation and allows promoter-proximal pausing [MNase]
Posted: 2015/04/04
Accession nr: GSE58956
Species: Drosophila melanogaster
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE58nnn/GSE58956/
Summary (begins on next line):
Promoter-proximal pausing of RNA polymerase II (Pol II) is a widespread in higher eukaryotes. Previous studies have shown that GAF is enriched at paused genes, but the role of GAF in pausing has not been well characterized on a genome-wide level. To investigate the role of GAF in pausing, we RNAi-depleted GAF from Drosophila S2 cells, and examined the effects on promoter-proximal polymerase. We confirmed the importance of GAF for pausing on the classic pause model gene Hsp70. To determine the dependence of pausing on GAF genome-wide, we assayed the levels of transcriptionally-engaged polymerase genome-wide using GRO-seq in control and GAF-RNAi cells. We found that promoter-proximal polymerase was significantly reduced on a subset of paused genes with GAF-bound promoters. There is a dramatic change in nucleosome distribution at genes with reduction in pausing upon GAF depletion and intergenic GAF binding sites in GAF knock-down, suggesting that GAF allows the establishment of pausing at these genes by directing nucleosome displacement off of the promoter. In addition, the insulator factor BEAF, BEAF-interacting protein Chriz, and transcription M1BP enrichment on unaffected genes suggests that redundant transcription factors or insulators protect other GAF-bound paused genes from GAF knock-down effects.

Data matrix URL defined? YES
Ready to fetch data? YES
Data fetched? NO
Protocol: gro-seq

----Summary for 200058957----
Title: GAGA Factor maintains promoters in nucleosome-free conformation and allows promoter-proximal pausing
Posted: 2015/04/04
Accession nr: GSE58957
Species: Drosophila melanogaster
Entry Type: GSEMatrix URL/FTP Link: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE58nnn/GSE58957/
Summary (begins on next line):
This SuperSeries is composed of the SubSeries listed below.

Data matrix URL defined? YES
Ready to fetch data? YES
Data fetched? NO
Protocol: gro-seq
\end{verbatim}

The following uses lastquery to fetch every pro-seq entry for Drosophila melanogaster:
\begin{verbatim}
user@computer-$ ./query_groseq_database.py -pt=pro-seq db findspecies "Drosophila melanogaster"
Found 5 elements that match "Drosophila melanogaster" given protocol(s) pro-seq
List of paths:
<cut>

user@computer-$ ./query_groseq_database.py -lq db fetchmatrices
Done with 200077607.
<cut>
Done with 200085747.
<cut>
Done with 200042397.
<cut>
Done with 200042117.
<cut>
Done with 200081649.
\end{verbatim}
\end{document}